### DISCUSSION TITLE
`schema plan` CLI command

### DISCUSSION THREAD

**Discussion by sjones6** [directus/directus#11976] [Feature Requests]
Similar to something like [`terraform plan`](https://www.terraform.io/cli/commands/plan) where you're able to diff the update against the infrastructure, it would be helpful to have a `schema plan` step that would diff a `schema.yaml` against the DB.

The use case is that we're using `directus schema apply --yes ./schema.yaml` as part of our CI pipelines—because this is not interactive, we don't have a way to review the changes and accept them before applying them to the DB.

The `apply` command already diffs the schema and DB state, so it would be just a matter of pulling that logic out of the `apply` command, running it separately and logging that out.

Happy to submit an PR if it's of interest to add to the team.

Thoughts?

**Comment by sjones6** [comment]
An alternative could be to add a `--dry-run` flag to the existing command:

```
directus schema apply --dry-run ./path/to/snapshot.yaml
```

**Reply by sjones6** [reply]
Here's an MR with the `--dry-run` flag approach: https://github.com/directus/directus/pull/12006

**Comment by rijkvanzanten** [comment]
Thanks! Great little improvement :)

**Comment by sjones6** [comment]
PR #12006 was accepted and merged (not yet released).
